### PR TITLE
Close #23: Add "Refund" function

### LINKED ISSUE #23: Add "Refund" function

### PR DESCRIPTION
**Pull request by UnimaginaryUnit**

### CODE REVIEW

File: GameToken.cs [hunk 1]
```diff
@@ -73,6 +73,24 @@ public Int32 MyBalance()
                 }
             } else Error.Throw("Operation denied");
         }
+
+        // Send GameTokens from recipient to sender.
+        // Sender should be present in the Game Developers white list.
+        // Can only be called from program present in the Game Developers white list.
+        public void Refund(Bytes recipient, Bytes sender, Int32 amount) {
+            if (WhiteListCheck(Info.Callers()[Info.Callers().Length-2]) && (WhiteListCheck(sender))) {
```
Comment (fomkin) [inline GameToken.cs]: What's a bullshit? What if callers stack length will be less than 3?

File: GameToken.cs [hunk 2]
```diff
@@ -73,6 +73,24 @@ public Int32 MyBalance()
                 }
             } else Error.Throw("Operation denied");
         }
+
+        // Send GameTokens from recipient to sender.
+        // Sender should be present in the Game Developers white list.
+        // Can only be called from program present in the Game Developers white list.
+        public void Refund(Bytes recipient, Bytes sender, Int32 amount) {
```
Comment (fomkin) [inline GameToken.cs]: From sender to recipient. Change arguments order.

File: GameToken.cs (hunk 1 above)
Comment (fomkin) [reply GameToken.cs]: Also add case when transaction sender is whitelisted.

File: GameToken.cs [hunk 3]
```diff
@@ -73,6 +73,31 @@ public Int32 MyBalance()
                 }
             } else Error.Throw("Operation denied");
         }
+
+        // Check if GameToken program was called from another program
+        private bool IsCalledFrom(Bytes address) {
+            if(Info.Callers().Length < 2) return false;
+            if(Info.Callers()[Info.Callers().Length-2] != address) return false;
+            return true;
```
Comment (Chicker) [inline GameToken.cs]: I think returning the `false` value by default is more secure.  We should return `true` only when we definitely know that this method was called by the `address`. In all other cases, we should return false.

**Pull request merged**